storage/src/main/java/zipkin2/storage/kafka/KafkaSpanStore.java

```diff
@@ -13,65 +13,59 @@
+import static zipkin2.storage.kafka.streams.TraceStoreSupplier.SERVICE_NAMES_STORE_NAME;
+import static zipkin2.storage.kafka.streams.TraceStoreSupplier.SPAN_IDS_BY_TS_STORE_NAME;
+import static zipkin2.storage.kafka.streams.TraceStoreSupplier.SPAN_NAMES_STORE_NAME;
+import static zipkin2.storage.kafka.streams.TraceStoreSupplier.TRACES_STORE_NAME;
 
 /**
- * Span Store based on Kafka Streams.
- *
- * This store supports all searches (e.g. findTraces, getTrace, getServiceNames, getSpanNames, and
- * getDependencies).
- *
- * NOTE: Currently State Stores are based on global state stores (i.e., all data is replicated on
- * every Zipkin instance with spanStoreEnabled=true).
+ * Span store backed by Kafka Stream State Stores.
+ * <p>
+ * These stores are currently supporting only single instance as there is not mechanism implemented
+ * for scatter gather data from different instances.
  */
-public class KafkaSpanStore implements SpanStore {
-  private static final Logger LOG = LoggerFactory.getLogger(KafkaSpanStore.class);
-  // Store names
-  final String tracesStoreName;
-  final String servicesStoreName;
-  final String dependenciesStoreName;
-  // Kafka Streams
+public class KafkaSpanStore implements SpanStore, ServiceAndSpanNames {
+  static final Logger LOG = LoggerFactory.getLogger(KafkaSpanStore.class);
+  // Kafka Streams Store provider
   final KafkaStreams traceStoreStream;
-  final KafkaStreams serviceStoreStream;
-  final KafkaStreams dependencyStoreStream;
-  // Span index
-  final SpanIndexService spanIndexService;
 
   KafkaSpanStore(KafkaStorage storage) {
-    tracesStoreName = storage.traceStoreName;
-    servicesStoreName = storage.serviceStoreName;
-    dependenciesStoreName = storage.dependencyStoreName;
     traceStoreStream = storage.getTraceStoreStream();
-    serviceStoreStream = storage.getServiceStoreStream();
-    dependencyStoreStream = storage.getDependencyStoreStream();
-    spanIndexService = storage.getSpanIndexService();
   }
 
-  @Override
-  public Call<List<List<Span>>> getTraces(QueryRequest request) {
+  @Override public Call<List<List<Span>>> getTraces(QueryRequest request) {
     try {
-      ReadOnlyKeyValueStore<String, List<Span>> traceStore =
-          traceStoreStream.store(tracesStoreName, QueryableStoreTypes.keyValueStore());
-      return new GetTracesCall(traceStore, spanIndexService, request);
+      ReadOnlyKeyValueStore<String, List<Span>> tracesStore =
+          traceStoreStream.store(TRACES_STORE_NAME, QueryableStoreTypes.keyValueStore());
+      ReadOnlyKeyValueStore<Long, Set<String>> traceIdsByTsStore =
+          traceStoreStream.store(SPAN_IDS_BY_TS_STORE_NAME, QueryableStoreTypes.keyValueStore());
+      return new GetTracesCall(tracesStore, traceIdsByTsStore, request);
     } catch (Exception e) {
-      LOG.error("Error getting traces", request, e);
+      LOG.error("Error getting traces. Request: {}", request, e);
```

Comment:
again probably this isn't best as it would be surprising as failures would only be visible in logs (not stats or call site)